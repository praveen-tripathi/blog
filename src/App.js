import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import "./styles.css";
import Home from "./components/Home";
import UserDetails from "./components/UserDetails";
import PostDetails from "./components/PostDetails";

export default function App() {
  return (
    <div className="container">
      <Router>
        <Switch>
          <Route exact path="/">
            <Home />
          </Route>
          <Route path="/user/:userId">
            <UserDetails />
          </Route>
          <Route path="/post/:postId">
            <PostDetails />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}
