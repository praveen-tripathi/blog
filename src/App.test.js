import React from "react";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import App from "./App";

Enzyme.configure({ adapter: new Adapter() });

describe("App Coponent", () => {
  it("should render app component", () => {
    const wrapper = shallow(<App />);

    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
