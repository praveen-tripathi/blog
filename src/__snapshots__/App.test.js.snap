// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Coponent should render app component 1`] = `
Array [
  <div
    className="container"
  >
    <BrowserRouter>
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <Home />
        </Route>
        <Route
          path="/user/:userId"
        >
          <UserDetails />
        </Route>
        <Route
          path="/post/:postId"
        >
          <PostDetails />
        </Route>
      </Switch>
    </BrowserRouter>
  </div>,
]
`;
